homeassistant/components/hassio/backup.py

```diff
@@ -635,7 +636,30 @@ def on_job_progress(data: Mapping[str, Any]) -> None:
             unsub()
 
     async def async_validate_config(self, *, config: BackupConfig) -> None:
-        """Validate backup config."""
+        """Validate backup config.
+
+        Replace the core backup agent with the hassio default agent.
+        """
+        core_agent_id = "backup.local"
+        create_backup = config.data.create_backup
+        if core_agent_id not in create_backup.agent_ids:
+            _LOGGER.debug("Backup settings don't need to be adjusted")
+            return
+
+        default_agent = await _default_agent(self._client)
+        _LOGGER.info("Adjusting backup settings to not include core backup location")
+        automatic_agents = [
+            agent_id if agent_id != core_agent_id else default_agent
+            for agent_id in create_backup.agent_ids
+        ]
+        await config.update(
+            create_backup=CreateBackupParametersDict(
+                agent_ids=automatic_agents,
+                include_addons=None,
```

Comment:
We should use the existing config settings for these other parameters. We don't know that we're migrating every time `async_validate_config` is called. It's only the agents that should be changed if needed.